Fix: `data-name` and `data-stack-name` with non-ASCII characters navigation issue by replacing `\W` with `\s`

KaoruBB opened the pull request:
## Description

This PR addresses the issue where navigation fails when using `data-name` and `data-stack-name` with non-ASCII characters (e.g., Japanese, Chinese, Korean). 
The previous implementation used the regex `\W` to remove non-word characters when generating IDs, which incorrectly stripped non-ASCII characters. 
This PR replaces `\W` with `\s` in the ID generation, removing only whitespace and preserving non-ASCII characters. 
This allows menu links to correctly navigate to sections with non-ASCII characters in their `data-name` and `data-stack-name`.
However, I'm a bit concerned that this change could introduce some side effects. 
Please let me know if you observe any issues.

## Changes

- Replaced `/\W/g` with `/\s/g` in the following locations:
    - `namedsection.id` generation
    - `firstChildSection.id` generation
    - `autoMenuLinks` generation
    - manual menu link `href` generation

## Testing

The issue was tested with `data-name` and `data-stack-name` containing Japanese characters. Navigation now works as expected.

KaoruBB committed:
replaces `\W` with `\s`

Martinomagnifico commented:
Could you send an example file that contains some of the offending characters? Thanks!

KaoruBB committed:
add demo files with Japanese data-(stack-)name

KaoruBB commented:
Hi Martino,
Thank you for your quick response and request for an example file.

I have added demo files that includes Japanese characters in `data-name` and `data-stack-name` attributes. 

Added files are 
- `demo-custom-jp.html`
- `custom-jp.md`

I hope this helps. Please let me know if you have any other questions.

Thanks again,
Kaoru

Martinomagnifico commented:
Should be fixed now.

KaoruBB committed:
Merge remote-tracking branch 'Martinomagnifico/master'

KaoruBB commented:
Thanks!!

Martinomagnifico commented:
Issue/PR can be closed, the changes (adjusted) are already in the 2.0.3 version.
